engine/execution/storehouse/in_memory_register_store_test.go

```diff
@@ -0,0 +1,635 @@
+	store := NewInMemoryRegisterStore(pruned, lastID)
+
+	// 10 <- A (X: 1, Y: 2) <- B (Y: 3)
+	blockA := unittest.IdentifierFixture()
+	regX := makeReg("X", "1")
+	regY := makeReg("Y", "2")
+	err := store.SaveRegisters(
+		pruned+1,
+		blockA,
+		lastID,
+		[]flow.RegisterEntry{regX, regY},
+	)
+	require.NoError(t, err)
+
+	blockB := unittest.IdentifierFixture()
+	regY3 := makeReg("Y", "3")
+	err = store.SaveRegisters(
+		pruned+2,
+		blockB,
+		blockA,
+		[]flow.RegisterEntry{regY3},
+	)
+	require.NoError(t, err)
+
+	val, err := store.GetRegister(pruned+2, blockB, regX.Key)
+	require.NoError(t, err)
+	require.Equal(t, regX.Value, val) // X is not updated in B
+
+	val, err = store.GetRegister(pruned+2, blockB, regY.Key)
+	require.NoError(t, err)
+	require.Equal(t, regY3.Value, val) // Y is updated in B
+
+	val, err = store.GetRegister(pruned+1, blockA, regY.Key)
+	require.NoError(t, err)
+	require.Equal(t, regY.Value, val) // Y's old value at A
+
+	_, err = store.GetRegister(pruned+2, blockB, unknownKey)
+	require.Error(t, err)
+	pe, ok := IsErrPruned(err)
+	require.True(t, ok)
+	require.Equal(t, pe.PrunedHeight, pruned)
+	require.Equal(t, pe.Height, pruned+2)
+
+	_, err = store.GetRegister(pruned+3, unittest.IdentifierFixture(), regX.Key)
+	require.Error(t, err)
+	require.ErrorIs(t, err, ErrNotExecuted) // unknown block
+}
+
+//  7. Given the following tree:
+//     Pruned <- A(X:1) <- B(Y:2)
+//     .......^- C(X:3) <- D(Y:4)
+//     GetRegister(D, X) should return 3
+func TestInMemoryRegisterStoreMultiForkOK(t *testing.T) {
+	t.Parallel()
+	pruned := uint64(10)
+	lastID := unittest.IdentifierFixture()
+	store := NewInMemoryRegisterStore(pruned, lastID)
+
+	// 10 <- A (X: 1) <- B (Y: 2)
+	//		^- C (X: 3) <- D (Y: 4)
+	blockA := unittest.IdentifierFixture()
+	blockB := unittest.IdentifierFixture()
+	blockC := unittest.IdentifierFixture()
+	blockD := unittest.IdentifierFixture()
+
+	require.NoError(t, store.SaveRegisters(
+		pruned+1,
+		blockA,
+		lastID,
+		[]flow.RegisterEntry{makeReg("X", "1")},
+	))
+
+	require.NoError(t, store.SaveRegisters(
+		pruned+2,
+		blockB,
+		blockA,
+		[]flow.RegisterEntry{makeReg("Y", "2")},
+	))
+
+	require.NoError(t, store.SaveRegisters(
+		pruned+1,
+		blockC,
+		lastID,
+		[]flow.RegisterEntry{makeReg("X", "3")},
+	))
+
+	require.NoError(t, store.SaveRegisters(
+		pruned+2,
+		blockD,
+		blockC,
+		[]flow.RegisterEntry{makeReg("Y", "4")},
+	))
+
+	reg := makeReg("X", "3")
+	val, err := store.GetRegister(pruned+2, blockD, reg.Key)
+	require.NoError(t, err)
+	require.Equal(t, reg.Value, val)
+}
+
+//  7. Given the following tree:
+//     Pruned <- A(X:1) <- B(Y:2), B is not executed
+//     GetUpdatedRegisters(C) should return ErrNotExecuted
+func TestInMemoryRegisterGetUpdatedRegisters(t *testing.T) {
+	t.Parallel()
+	pruned := uint64(10)
+	lastID := unittest.IdentifierFixture()
+	store := NewInMemoryRegisterStore(pruned, lastID)
+
+	// 10 <- A (X: 1) <- B (Y: 2)
+	//		^- C (X: 3) <- D (Y: 4)
```

Comment:
doesn't look like there is a `C` in this test. should there be, or can this line be removed?